x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedJobValidator.java

```diff
@@ -34,10 +34,15 @@ public static void validate(DatafeedConfig datafeedConfig, Job job) {
         }
 
         DelayedDataCheckConfig delayedDataCheckConfig = datafeedConfig.getDelayedDataCheckConfig();
-        TimeValue delayedDataCheckWindow =  delayedDataCheckConfig.getCheckWindow();
         TimeValue bucketSpan = analysisConfig.getBucketSpan();
-        if (delayedDataCheckConfig.isEnabled()
-            && delayedDataCheckWindow.equals(DelayedDataCheckConfig.DEFAULT_DELAYED_DATA_WINDOW) == false) {
+        if (delayedDataCheckConfig.isEnabled()) {
+            checkValidDelayedDataCheckConfig(bucketSpan, delayedDataCheckConfig);
+        }
+    }
+
+    private static void checkValidDelayedDataCheckConfig(TimeValue bucketSpan, DelayedDataCheckConfig delayedDataCheckConfig) {
+        TimeValue delayedDataCheckWindow =  delayedDataCheckConfig.getCheckWindow();
+        if (delayedDataCheckWindow.equals(DelayedDataCheckConfig.DEFAULT_DELAYED_DATA_WINDOW) == false) {
```

Comment:
Not sure it's worth splitting path here. The default value should also be a valid one so why not ignore it?

Reply:
What if the `bucket_span` is larger than `DelayedDataCheckConfig.DEFAULT_DELAYED_DATA_WINDOW` ?

As for the `window` spanning too many buckets, that is not possible with a 2 hour default and requiring `bucket_span` to be measured in seconds.

Reply:
Hm, good point. But then we get in a situation where we cannot distinguish really between a default config and a user one set to 2 hours. Let's discuss this futher offline.

Reply:
We discussed this offline. 

The idea here is inherently broken. We should instead allow a `null` value for the window and then calculate the appropriate default based off of the `bucket_span` and other criteria. 

Working on adjusting this now.